File: code-samles/sample.lua
-- Lua — syntax sampler
--[[ block comment ]]
local M = {}
local Color = { RED = 1, GREEN = 2, BLUE = 3 }

local function distance(a, b)
    local dx, dy = a.x - b.x, a.y - b.y
    return (dx * dx + dy * dy) ^ 0.5
end

M.Point = {
    __index = function(t, k)
        return rawget(t, k) or 0
    end,
}
function M.Point:new(x, y)
    local o = setmetatable({ x = x or 0, y = y or 0 }, self)
    return o
end

local fmt = string.format
local msg = ([[color: %s → %.2f]]):gsub("color", "color"):gsub("%s", " ")
local p = M.Point:new(0, 1)
local q = M.Point:new(2, 3)

local ok, res = pcall(function()
    local d = distance(p, q)
    return { label = "dist", value = d, color = Color.RED }
end)

for k, v in pairs(ok and res or {}) do
    print(fmt("%s = %s", k, tostring(v)))
end

-- pattern match (not full regex)
local s = "hello-world"
if s:match("^hello[%s%-]*world$") then
    print(msg:format("RED", distance(p, q)))
end

return M

File: code-samles/sample.py
# Python — syntax sampler
from __future__ import annotations
import re
from dataclasses import dataclass
from enum import Enum
from typing import Generic, TypeVar

# TODO(yourname): tighten exception handling
T = TypeVar("T")


class Color(Enum):
    RED = 1
    GREEN = 2
    BLUE = 3


@dataclass(slots=True, frozen=True)
class Point(Generic[T]):
    x: T
    y: T


def memoize(fn):
    cache = {}

    def wrapper(*args, **kwargs):
        key = args, tuple(sorted(kwargs.items()))
        if key in cache:
            return cache[key]
        cache[key] = fn(*args, **kwargs)
        return cache[key]

    return wrapper


@memoize
def distance(a: Point[float], b: Point[float]) -> float:
    dx, dy = a.x - b.x, a.y - b.y
    return (dx**2 + dy**2) ** 0.5


async def fetch(url: str) -> bytes:
    import aiohttp

    async with aiohttp.ClientSession() as s:
        async with s.get(url) as r:
            return await r.read()


pattern = re.compile(r"^hello[\s\-]*(world)?$", re.IGNORECASE | re.M)
names = ["alice", "bob", "éva"]
squares = {n: i * i for i, n in enumerate(names)}

try:
    msg = f"{Color.RED.name} → {distance(Point(0.0, 1.0), Point(2.0, 3.0)):.2f}"
    assert pattern.search("Hello  - world")
except (AssertionError, ValueError) as e:
    print(f"error: {e!r}")
else:
    with open("out.txt", "w", encoding="utf-8") as f:
        f.write(msg + "\n")
finally:
    print(*(x for x in range(3)), end="!\n")
